File: cpp/19/19.cpp
/*
 * Created Date: Sunday, August 2nd 2020, 7:17:10 pm
 * Author: Chirag Sachdev
 * LinkedIn: https://www.linkedin.com/in/chiragsachdev/
 * Github: https://github.com/chiragsachdev/
 */


#include <iostream>
using namespace std;

int main()
{
	int ctr;
	int curr;
	int year;
	int month;
	int n_days;

	curr = 1;
	ctr = 0;
	for (year = 1901; year <= 2000; year++)
	{
		for (month = 1; month <= 12; month++)
		{
			if ((year % 4 == 0) && (month == 2))
				n_days = 29;
			else
			{
				switch (month)
				{
					case (1):	n_days = 31;	break;
					case (2):	n_days = 28;	break;
					case (3):	n_days = 31;	break;
					case (4):	n_days = 30;	break;
					case (5):	n_days = 31;	break;
					case (6):	n_days = 30;	break;
					case (7):	n_days = 31;	break;
					case (8):	n_days = 31;	break;
					case (9):	n_days = 30;	break;
					case (10):	n_days = 31;	break;
					case (11):	n_days = 30;	break;
					case (12):	n_days = 31;	break;
				}
			}
			curr = curr + (n_days - 28);
			if (curr > 6)
				curr %= 7;
			if (curr == 6)
				ctr++;			
		}
	}

	cout << ctr << endl;
	return (0);
}

File: Python3/19/19.py
#!/usr/bin/env python3
#==============================================================================#
# Created Date: Sunday, August 2nd 2020, 5:51:13 pm                            #
# Author: Chirag Sachdev                                                       #
# LinkedIn: https://www.linkedin.com/in/chiragsachdev/                         #
# Github: https://github.com/chiragsachdev/                                    #
#==============================================================================#


mapper = {
	1:31,
	2:28,
	3:31,
	4:30,
	5:31,
	6:30,
	7:31,
	8:31,
	9:30,
	10:31,
	11:30,
	12:31
}

days = [
	'monday',
	'tuesday',
	'wednesday',
	'thursday',
	'friday',
	'saturday',
	'sunday'
]

ctr = 0
total = 0
curr = 1
for year in range(1901,2001):
	for month in range(1,13):
		if (year % 4 == 0) and month == 2:
			n_days = 29
		else:
			n_days = mapper[month]
		curr = curr + (n_days - 28)
		if curr > 6:
			curr = curr % 7
		if curr == 6:
			ctr += 1
print(ctr)
